doc/samples/SqlConfigurableRetryLogic_OpenConnection.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+// <Snippet1>
+using Microsoft.Data.SqlClient;
+
+/// Detecting retriable exceptions is a vital part of the retry pattern.
+/// Before applying retry logic it is important to investigate exceptions and choose a retry provider that best fits your scenario.
+/// First, log your exceptions and find transient faults.
+/// The purpose of this sample is to illustrate how to use this feature and the condition might not be realistic.
+class RetryLogicSample
+{
+    private const string DefaultDB = "Northwind";
+    private const string CnnStringFormat = "Server=localhost; Initial Catalog={0}; Integrated Security=true; pooling=false;";
+    private const string DropDatabaseFormat = "DROP DATABASE {0}";
+
+    // For general use
+    private static SqlConnection s_generalConnection = new SqlConnection(string.Format(CnnStringFormat, DefaultDB));
+
+    /*static*/ void Main(string[] args)
+    {
+        // 1. Define the retry logic parameters
+        var options = new FloatingRetryLogicOption()
+        {
+            NumberOfTries = 5,
+            MaxTimeInterval = TimeSpan.FromSeconds(20),
+            DeltaTime = TimeSpan.FromSeconds(1)
+        };
+
+        // 2. Create a retry provider
+        var provider = SqlConfigurableRetryFactory.CreateExponentialRetryProvider(options);
+
+        // define the retrying event to report the execution attempts
+        provider.Retrying += (object s, SqlRetryingEventArgs e) =>
+            {
+                int attempts = e.RetryCount + 1;
+                Console.ForegroundColor = ConsoleColor.Yellow;
+                Console.WriteLine($"attempt {attempts} - current delay time:{e.Delay} \n");
+                Console.ForegroundColor = ConsoleColor.DarkGray;
+                if (e.Exceptions[e.Exceptions.Count - 1] is SqlException ex)
+                {
+                    Console.WriteLine($"{ex.Number}-{ex.Message}\n");
+                }
+                else
+                {
+                    Console.WriteLine($"{e.Exceptions[e.Exceptions.Count - 1].Message}\n");
+                }
+
+                // This is not a good practice to do time-consuming tasks inside the retrying event and blocking the running task.
+                // Use the parallel programming patterns to mitigate it.
+                if (e.RetryCount == provider.RetryLogic.NumberOfTries - 1)
+                {
+                    Console.WriteLine("This is the last chance to execute the command before throwing the exception.");
+                    Console.WriteLine("Press Enter when you're ready:");
+                    Console.ReadLine();
+                    Console.WriteLine("continue ...");
+                }
+            };
+
+        // Open the general connection.
+        s_generalConnection.Open();
+
+        try
+        {
+            // Assume the database is creating and other services are going to connect to it.
```

Comment:
This comment is unclear as to how it relates to what happens in `RetryConnection`.

Reply:
I found it confusing because it says "assume the database is created" but then in the method, we haven't created the database yet. I think something simpler like "Induce a transient error to initiate retry attempts" would be enough and then the comments in the method will explain how that happens.